### DISCUSSION TITLE
Proposta de criação de servidor para hospedar a plataforma de submissão de palestras Pretalx

### DISCUSSION THREAD

**Discussion by renan-asantos** [apyb/comunidade#111] [Geral]
Opa pessoal, tudo certo?

Um dia desses eu e a @anadulce estávamos conversando sobre os eventos de Python que ocorrem aqui no Brasil e que alguns deles já utilizaram a plataforma de submissão de palestras [Pretalx](https://pretalx.com/p/about/), vide PyBR 2x e PyNE 1x.

O pretalx é um software open-source que pode ser utilizado de duas maneiras: 
1 - pagando pelo uso da plataforma hospedada nos servidores deles
2 - subir uma instância para hospedar a própria versão da plataforma. 

Na opção 1 o custo depende de duas variáveis do evento: *quantidade esperada de pessoas* e o *preço do ingresso*.
![image](https://github.com/apyb/comunidade/assets/61845573/e8ee1a4a-018a-4773-b189-b7817bdf72ef)
A variável do preço do ingresso sempre fica entre 0-100 euros, visto que os ingressos no Brasil não costumam ultrapassar 100 euros e considerando o público da Python Brasil que costuma ultrapassar os 500 participantes, o preço pelo uso da plataforma seria de 449 Euros *por evento*.
O pretalx fornece um desconto de 25% na hospedagem caso seja um evento de comunidade sem fins lucrativos e mantido por voluntários. Assim, o preço final do uso da plataforma pelo site do pretalx gira em torno de 336,75 euros (R$ 1819,66 na cotação atual).

Até o momento, os eventos de Python do Brasil seguiram com a primeira opção para não ter que lidar com a complexidade de subir e manter um servidor, mas na conversa com a @anadulce pensamos que seria vantajoso ter uma hospedagem própria. Isso permitiria diminuir o custo em cada evento, proporcionar uma padronização na plataforma de submissão de atividades, além de acessibilizar a eventos menores, que não teriam condições de arcar com o custos da plataforma original. Sabemos que o acesso ao pretalx ajuda a organização a receber propostas mais bem estruturadas, e que isso impacta diretamente na qualidade do evento final, portanto é uma contribuição para toda a comunidade.

Acredito que já validamos a experiência com a ferramenta o suficiente para considerá-la uma opção a longo prazo para a comunidade, e que nesse momento, seria vantajoso tanto em termos de custo quanto de organização, migrar para a hospedagem própria deste serviço.

O pretalx é feito em python e django, eu mesmo já subi localmente, mexi e testei. Acredito que seja viável subir uma instância em um CloudRun ou algum outro recurso de baixa manutenção, que nos forneça segurança para o uso, sem demandar grandes esforços de pessoas para mantê-lo.

Gostaria de propor um grupo de trabalho voltado a pesquisar a viabilidade e custo de internalizar o pretalx para uso da comunidade Python no Brasil, e mensurar qual seria o esforço de manutenção.

**Comment by anadulce** [comment]
Reforço que considero uma grande contribuição para com a comunidade e as organizações dos eventos, e que acho que temos maturidade o suficiente para colocar em prática.

**Comment by PySulBR** [comment]
Opa pessoal, adiciono a PySul como uma comunidade que gostaria de usufruir dessa melhoria também, e deixo o @gabrielecker em contato para ajudarmos a viabilizar isso.

Talita Rossar, Co-organizadora da PySul 2024.

**Comment by naanadr** [comment]
Obrigada @renan-assis-santos pela pré-análise e pela proposta 😄 

Será avaliado outras ferramentas parecidas? Pergunto isso, porque a Python Nordeste desse ano está utilizando o [even3](https://www.even3.com.br/como-funciona/trabalhos-cientificos/), seria interessante conversar com o pessoal para ver como está sendo a experiência com esse site (cc @itepifanio e @ryllari)

Vocês conseguem ter uma ideia de quanto seria o custo medio de hospedagem do Pretalx, ou pretendem investigar isso melhor no GT? 🤔 

Além dessas dúvidas, gostaria de salientar a importância de validar nesse GT o custo e a complexidade de hospedagem e de manter o Pretalx na possível infra da APyB. Porque hoje já estamos tentando reduzir gastos com servidor, por exemplo, avaliando alternativas de hospedagem para o site de associados.

Fazer um grupo de trabalho levando esses pontos em consideração, e sem esquecer de também trazer a discussão de quem ficaria responsável por manter o sistema (caso venha a ser implementado) é muito bem vindo! 
A diretoria fica a disposição para auxiliar no que for preciso ❤️ .

**Reply by renan-asantos** [reply]
@naanadr enquanto fiz parte da org da python brasil de 2022 fiz um estudo sobre algumas ferramentas que poderíamos usar na época e no final concluí que o pretalx era a melhor delas ([link da issue que fiz a análise](https://github.com/pythonbrasil/pybr2022-org/issues/48)). Adiciono também que essa foi uma análise antes de usar a plataforma, depois do final do evento poderia adicionar vários outras vantagens.

Vendo o [even3](https://www.even3.com.br/como-funciona/trabalhos-cientificos/) me pareceu ser uma boa alternativa também, mas vejo que exigiria uma análise mais detalhada de prós e contras. Além de que se a Python Nordeste está usando apenas esse ano diria que faltam experiências que validem a ferramenta para o tipo de evento que fazemos, já que o pretalx foi validado em duas edições da Python Brasil e vários outros eventos de python internacionais utilizam também.

@itepifanio e @ryllari, vocês possuem alguma análise feita para decidir a escolha do even3 na Python Nordeste? Se não, conseguiriam comentar a experiência de vocês até agora com a plataforma?

Eu abri a discussão primeiro para entender com a comunidade e com vocês da APyB se a ideia faz sentido, com certeza teria que ser bem pensado o custo, complexidade de hospedagem e de manter a plataforma na infra própria. Na questão do custo, vejo que diminuiria pensando em toda a comunidade de eventos python, sendo que ao invés do custo de hospedagem em euros, existiria apenas o custo de manter o servidor próprio.

Com certeza consideraríamos as pessoas que seriam responsáveis por manter o sistema, mas só entraria nesse detalhe se a ideia fosse bem aceita e apoiada.

**Reply by itepifanio** [reply]
@renan-assis-santos @naanadr @biancarosa segue [a proposta comercial da Even3](https://drive.google.com/file/d/16af9atDw6koxrlQFdgjvSlaTMXpeuQyc/view). A plataforma custou R$ 808,00 e 9,5% por ingresso. O que mais nos chamou atenção foi a possibilidade de realizar o pagamento após o fim do evento, através dos valores recebidos dos ingressos, além disso você tem um local só que agrega submissões e venda de ingressos (além de sites se for necessário).

Eles tem um custo de "atendimento vip" de 1668,00R$ que decidimos não pagar. Sentimos um pouco a falta desse atendimento porque o sistema deles não é muito flexível, por exemplo, ao cadastrar uma pergunta você não pode reordenar ou mudar o tipo de pergunta após o cadastro, tivemos que recadastrar tudo por conta de um erro, além disso eles tem uns campos obrigatórios que não necessariamente fazem muito sentido pro seu evento e sempre fica a dúvida se aquele campo é realmente obrigatório. O suporte deles demora em torno de 48h para responder.

O uso da Even3 para a Python Nordeste não é novo, foi utilizado em 2019 também. Até agora, tirando alguns problemas de usabilidade pontuais, achamos que ele tem um bom custo benefício. Durante nossa análise vimos a Even3 e o Pretalx, ficamos nesse mesmo dilema de subir uma instância do Pretalx, mas a conveniência do serviço, o preço, e evitar trabalho de manutenção nos fez nos optar pelo Even3.

**Comment by biancarosa** [comment]
Eu voto pelo uso do pretalx pelas comunidades pela sua reputação, por ser open source, e pelo seu uso nas comunidades internacionais. Não comparei feature by feature com outras plataformas como event3. 

O @flaviocpontes deu uma olhada geral na arquitetura e parece que uma máquina EC2, um PostresSQL e um Redis seriam suficientes na AWS. O custo de uma infra assim não costuma ser alto - comparado com os 500 eur que estamos pagando ano a ano -, mas precisa de uma certa manutenção. 

Minha proposta seria tentar subir o pretalx como piloto para um evento (ex pybr) e a pybr absorveria esse custo. Criamos a conta nós mesmos e pedimos o reembolso etc. A APyB pode conversar internamente e decidir se quer oferecer essa estrutura pros eventos depois, nem que seja cobrando um valor fixo $$ de cada evento. Eu prefiro pagar 1000 BRL à APyB pra ajudar a custear essa infra a pagar 500 eur pra plataforma uma vez que a realidade financeira de eventos como a pycon é muito diferente.

**Reply by PySulBR** [reply]
opa @biancarosa, podemos unir forças entre PyBR24 e PySul24, assim, dividiriamos o valor também.
O que você acha de marcarmos uma conversa entre orgs e traçarmos o plano?

Talita Rossari

**Reply by renan-asantos** [reply]
Achei ótima a ideia @biancarosa!

Pensei no GCP com um Cloud Run e um Postgres tbm, mas só ideia inicial mesmo e que eu manjava mais. Ainda não cheguei a avaliar aonde seria mais recomendado (menor custo, facil manutenção, etc).

O único contra que consigo pensar (e vejo até como aceitável) é se por algum motivo não mantermos esse piloto, assim as submissões, histórico e dados no pretalx da Python BRasil 2024 seriam perdidos, visto que não manteriam um servidor na AWS só para manter o histórico.

Se quiserem manter esse tópico de discussão aberto para pensarmos juntos como fazer isso, qual servidor usar, possíveis responsáveis etc eu contribuiria.
Se forem transpor essa discussão para algum fórum da Python Brasil eu gostaria de ser incluído para poder contribuir também

**Reply by biancarosa** [reply]
Hoje o @flaviocpontes ficou de tocar isso na org da PyBR - vamos juntar um grupo de trabalho aqui!

**Comment by gabubellon** [comment]
Pessoas lindas, estou fazendo a lição de casa e vendo o que perdi por aqui dando meus comentário onde acho interessante.

Primeiro, queria me deixar a disposição de qualquer organizador (seja a PyBrasil ou PySul) para ajudar a colocar no ar e dar manutenção nessa possivel prova de conceito e afins.

Tenho total interesse uma vez que isso está diretamente ligado aos dados dos eventos e assunto que tenho muito interesse aqui na comunidade

Segundo ponto, podemos usar isso como moeda de patrocínio não ? O não já temos, nada impede da gente bater na porta da AWS, Google e Digital Ocean e pedir um patrocínio do evento em forma de créditos e afins

Além do mais sei que existem os usos de no-profit para Google Workspace e afins, talvez a empresa que fez o contrato no-profit da Google para APyB  possa orientar e talvez ajudar (posso dar uma olhada nisso)


Mas usa coisa que pode ser feita agora (talvez) já seja criar um repo e um projeto aqui dentro da estrutura da apyb ou da python brasil pra compartilhar info e termos mais gente se interessando em ajudar

@renan-assis-santos @biancarosa  o que acham ?

**Reply by biancarosa** [reply]
excelente! acho que um grupo de trabalho envolvendo eu, o @flaviocpontes, o @renan-assis-santos, a Talita (@PySulBR) e você @gabubellon pode ser interessante. eu não acho que consigo ser muito hands on devido ao trabalho da pybr, mas gostaria de estar por dentro da solução e colaborar aonde for necessário.

**Reply by biancarosa** [reply]
imagino que a melhor forma seja a gente se juntar num grupo do telegram mesmo e ver um horário que funciona pra fazermos um sync de 15 mins e iniciar o projeto.

**Reply by naanadr** [reply]
podem me colocar no grupo também, não consigo participar muito ativamente do GT, mas estarei disponível para tirar dúvidas sobre o que temos hoje na APyB.

**Reply by renan-asantos** [reply]
Boa, faz sentido pra mim. Conseguem me passar o @ de vcs do Telegram pra eu criar um grupo e iniciarmos algumas ideias sobre?
@biancarosa @naanadr @gabubellon @flaviocpontes @PySulBR

**Reply by PySulBR** [reply]
opa @gabrielecker irá tocar do lado da PySul
O telegram dele é: ```@gabsecker```

**Reply by gabubellon** [reply]
@renan-assis-santos é @gabubellon  mesmo :D

**Reply by naanadr** [reply]
@renan-assis-santos meu telegram é o mesmo @ do Github 😅

**Reply by tiidadavena** [reply]
@julianabrlm no Telegram: posso ajudar em testes de usabilidade e ajustes na UI.

**Comment by rennerocha** [comment]
Eu acho que seria ótimo ter uma instância do Pretalx disponível para qualquer evento da comunidade utilizar. O único porém é que estou vendo muita discussão e ideias a respeito do custo de infraestrutura - que é válido, mas acho mais simples de se resolver (conseguir apoio de algum provedor de cloud não é algo impossível de se conseguir) - mas pouca discussão sobre quem manterá essa infraestrutura.

Talvez para implementar esse servidor, poderíamos iniciar a discussão da possibilidade de remunerar alguém (ao invés de contar apenas com voluntárias) para configurar e monitorar esse serviço (e talvez outros, como o site de associados) para garantir a estabilidade dele. pretalx é algo que não pode ficar fora do ar muito tempo por algum problema aleatório e dependendo de uma pessoa voluntária tenha tempo para resolver. Isso pode causar um impacto mto grande na organização de eventos.

**Reply by rougeth** [reply]
A melhor opção é uma pessoa ficar responsável pelo projeto, estilo um big kahuna. Essa pessoa ficaria responsável pelo projeto, pela infra, operação, voluntários e custos. Uma vez com o custo definido, isso entra como custo para a Python Brasil.

Dessa forma, não aumentamos a carga de trabalho da diretoria da APyB e aumentamos a chance do projeto ser bem sucedido.

Contratar alguém me parece mais difícil burocraticamente, e provável do custo ficar maior do que pagar a ferramenta diretamente.

**Reply by naanadr** [reply]
@rennerocha eu havia mencionado anteriormente sobre esse ponto de quem ficaria responsável por manter o projeto. Mas não quis bater novamente nessa tecla enquanto não temos uma ideia de qual a complexidade de ter o Pretalx e enquanto o GT não começa a funcionar de verdade. 

Sobre remuneração, por enquanto não seria nada viável. E claro, só integraremos o Pretalx na infra da APyB quando estiver definido quem seriam as pessoas responsáveis por cuidar disso. Mas antes de tudo, seria interessante o GT se iniciar 😄 o que temos por enquanto são apenas suposições.

**Reply by tiidadavena** [reply]
Fazendo uma analogia bastante grosseira, acredito que essa instância do Pretalx seria algo open-source que nem o [Speakerfight](https://speakerfight.com/), que era mantido por core mantainers da comunidade. E a passagem de bastão poderia ser tal qual o onboarding das equipes organizadoras de evento a cada virada de ano; aproveito esse ponto para reforçar que uma documentação **robusta** deve fazer parte dessa iniciativa da instância.

**Reply by rennerocha** [reply]
> A melhor opção é uma pessoa ficar responsável pelo projeto, estilo um big kahuna. Essa pessoa ficaria responsável pelo projeto, pela infra, operação, voluntários e custos. Uma vez com o custo definido, isso entra como custo para a Python Brasil.

Parece uma ótima ideia ter uma pessoa responsável (que não precisa ser alguém da Diretoria eleita) para ser quem responde por esse projeto (não necessariamente quem vai botar a mão na massa de manter tudo funcionando) e vai ficar lendo todos nossos palpites aqui e decidir o que serve e o que não serve ignorar 😄. E com o tempo essa pessoa pode ser substituída. Acho que isso é a maneira de fazer isso funcionar e se manter no longo prazo.

> Contratar alguém me parece mais difícil burocraticamente, e provável do custo ficar maior do que pagar a ferramenta diretamente.

A APyB pode fazer um contrato de serviço com alguma empresa (alguém PJ) para ter o monitoramento e manutenção do sistema e isso entrar como um custo recorrente. Não é necessário uma contratação CLT de alguém (aí sim, isso seria super burocrático).

Sobre o custo, aí vai depender do uso do sistema. Se só for utilizado para a Python Brasil, realmente é mais prático usar o deles que já está rodando (apesar de caro). Se todos os eventos regionais, meetups locais, etc. usarem, aí o custo é diluído e a comunidade toda se beneficia!

**Comment by tiidadavena** [comment]
Outro ponto que quero trazer para essa instância é que no seu princípio de MVP consiste em um fluxo único de avaliação das propostas e.g. revisão cega das propostas, se pessoas da organização podem submeter palestras etc. Levanto esse tópico pois os critérios, e, em especial, o processo de avaliação são diferentes para cada evento; então considero salutar que todos os eventos que fizerem uso da instância tenham o fluxo de avaliação de propostas padronizado.

A sugestão acima considerando o cenário mais simples de implementação; sem features dedicadas para cada customização de fluxo de avaliação de palestras.
